# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/VideoSink.h
# Commit: 1c45b135318d
# Full Hash: 1c45b135318d8cb519b8d58af72ab864fe6bfd62
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-11-14 17:12:31
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1172394 - Simplify MediaSink somewhat. r=padenot
#   
#   This patches does several minor things:
#   - Moves SetSink (from setSinkid) to automatic coalescing of multiple calls
#     through a Canonical/Mirror setup instead of a manual atomic counter.
# ==============================================================================

diff -r c57c41e8c39e -r 1c45b135318d dom/media/mediasink/VideoSink.h
--- a/dom/media/mediasink/VideoSink.h	Wed Nov 13 08:55:39 2019 +0000
+++ b/dom/media/mediasink/VideoSink.h	Wed Nov 13 08:55:54 2019 +0000
@@ -32,10 +32,6 @@
             MediaQueue<VideoData>& aVideoQueue, VideoFrameContainer* aContainer,
             FrameStatistics& aFrameStats, uint32_t aVQueueSentToCompositerSize);
 
-  const PlaybackParams& GetPlaybackParams() const override;
-
-  void SetPlaybackParams(const PlaybackParams& aParams) override;
-
   RefPtr<EndedPromise> OnEnded(TrackType aType) override;
 
   TimeUnit GetEndTime(TrackType aType) const override;
@@ -52,6 +48,8 @@
 
   void SetPlaying(bool aPlaying) override;
 
+  double PlaybackRate() const override;
+
   void Redraw(const VideoInfo& aInfo) override;
 
   nsresult Start(const TimeUnit& aStartTime, const MediaInfo& aInfo) override;
